lib/ts/recipe/emailverification/api/implementation.ts

```diff
@@ -1,112 +1,118 @@
+            }
+            return res;
         },
 
         isEmailVerifiedGET: async function ({
-            options,
             userContext,
-        }: {
-            options: APIOptions;
-            userContext: any;
+            session,
         }): Promise<
             | {
                   status: "OK";
                   isVerified: boolean;
               }
             | GeneralErrorResponse
         > {
-            let session = await Session.getSession(
-                options.req,
-                options.res,
-                { overrideGlobalClaimValidators: () => [] },
-                userContext
-            );
-
             if (session === undefined) {
                 throw new Error("Session is undefined. Should not come here.");
             }
 
-            let userId = session.getUserId();
-
-            let email = await options.config.getEmailForUserId(userId, userContext);
+            await session.fetchAndSetClaim(EmailVerificationClaim, userContext);
+            const isVerified = await session.getClaimValue(EmailVerificationClaim, userContext);
 
             return {
                 status: "OK",
-                isVerified: await options.recipeImplementation.isEmailVerified({ userId, email, userContext }),
+                isVerified: isVerified === true,
             };
         },
 
         generateEmailVerifyTokenPOST: async function ({
             options,
             userContext,
-        }: {
-            options: APIOptions;
-            userContext: any;
+            session,
         }): Promise<{ status: "OK" | "EMAIL_ALREADY_VERIFIED_ERROR" } | GeneralErrorResponse> {
-            let session = await Session.getSession(
-                options.req,
-                options.res,
-                { overrideGlobalClaimValidators: () => [] },
-                userContext
-            );
-
             if (session === undefined) {
                 throw new Error("Session is undefined. Should not come here.");
             }
 
-            let userId = session.getUserId();
+            const userId = session.getUserId();
 
-            let email = await options.config.getEmailForUserId(userId, userContext);
-
-            let response = await options.recipeImplementation.createEmailVerificationToken({
+            const emailInfo = await EmailVerificationRecipe.getInstanceOrThrowError().getEmailForUserId(
                 userId,
-                email,
-                userContext,
-            });
+                userContext
+            );
 
-            if (response.status === "EMAIL_ALREADY_VERIFIED_ERROR") {
-                logDebugMessage(`Email verification email not sent to ${email} because it is already verified.`);
-                return response;
-            }
+            if (emailInfo.status === "UNKNOWN_USER_ID_ERROR") {
+                return {
+                    status: "GENERAL_ERROR",
+                    message: "UNKNOWN_USER_ID_ERROR",
```

Comment:
Maybe we should throw an error here -> cause otherwise the user will see this error message on their UI. This should come here only if there is a programming error, so better to throw a generic 500 error?